internal/storage/rocksdb/store.go

```diff
@@ -410,49 +508,95 @@ func (rdb *rocksDB) SaveChanges(changes []*serverpb.ChangeRecord) (uint64, error
 }
 
 type iter struct {
-	iterOpts storage.IterationOptions
-	rdbIter  *gorocksdb.Iterator
+	iterOpts    storage.IterationOptions
+	rdbIter     []*gorocksdb.Iterator
+	currentIter int
 }
 
 func (rdb *rocksDB) newIter(iterOpts storage.IterationOptions) *iter {
 	readOpts := rdb.opts.readOpts
-	it := rdb.db.NewIterator(readOpts)
 
+	it := rdb.db.NewIterator(readOpts)
 	if sk, prsnt := iterOpts.StartKey(); prsnt {
 		it.Seek(sk)
 	} else {
 		it.SeekToFirst()
 	}
-	return &iter{iterOpts, it}
+
+	itTTL := rdb.db.NewIteratorCF(readOpts, rdb.ttlCF)
```

Comment:
Given the slightly complex logic, please ensure we have at least the following test cases around iteration.

1. keyspace with both TTL and normal keys
2. keyspace with only TTL keys
3. keyspace with only normal keys
4. keyspace with only TTL keys but all expired
5. keyspace with only TTL keys but non-matching prefix, but valid TTL.